tests/model_registry/conftest.py

```diff
@@ -22,186 +24,184 @@
 
 from tests.model_registry.constants import (
     MR_OPERATOR_NAME,
-    MR_INSTANCE_NAME,
-    DB_RESOURCES_NAME,
+    MR_INSTANCE_BASE_NAME,
+    DB_BASE_RESOURCES_NAME,
     MODEL_REGISTRY_DB_SECRET_STR_DATA,
-    MODEL_REGISTRY_DB_SECRET_ANNOTATIONS,
     OAUTH_PROXY_CONFIG_DICT,
-    MODEL_REGISTRY_STANDARD_LABELS,
     ISTIO_CONFIG_DICT,
+    DB_RESOURCE_NAME,
+    MR_INSTANCE_NAME,
 )
 from tests.model_registry.rest_api.utils import ModelRegistryV1Alpha1
 from utilities.constants import Labels, Protocols
 from tests.model_registry.utils import (
     get_endpoint_from_mr_service,
     get_mr_service_by_label,
-    get_model_registry_deployment_template_dict,
-    get_model_registry_db_label_dict,
     wait_for_pods_running,
+    get_mr_service_objects,
+    get_mr_pvc_objects,
+    get_mr_secret_objects,
+    get_mr_deployment_objects,
+    get_model_registry_objects,
 )
 from utilities.constants import DscComponents
 from model_registry import ModelRegistry as ModelRegistryClient
-from semver import Version
 from utilities.general import wait_for_pods_by_labels
+from utilities.infra import get_data_science_cluster
 
 LOGGER = get_logger(name=__name__)
 
-MIN_MR_VERSION = Version.parse(version="2.20.0")
-
 
 @pytest.fixture(scope="class")
 def model_registry_namespace(updated_dsc_component_state_scope_class: DataScienceCluster) -> str:
     return updated_dsc_component_state_scope_class.instance.spec.components.modelregistry.registriesNamespace
 
 
+# @pytest.fixture(scope="class")
+# def model_registry_namespace(request: pytest.FixtureRequest, admin_client: DynamicClient) -> str:
+#     param_namespace = getattr(request, "param") or py_config["model_registry_namespace"]
+#     dsc = get_data_science_cluster(client=admin_client)
+#     namespace = dsc.instance.spec.components.modelregistry.registriesNamespace
+#     LOGGER.info(f"Model Registry namespace from DSC: {namespace}, from test: {param_namespace}")
+#     assert param_namespace == namespace
+#     return namespace
+
+
 @pytest.fixture(scope="class")
 def model_registry_db_service(
+    request: pytest.FixtureRequest,
     pytestconfig: Config,
     admin_client: DynamicClient,
-    model_registry_namespace: str,
     teardown_resources: bool,
+    model_registry_namespace: str,
     is_model_registry_oauth: bool,
-) -> Generator[Service, Any, Any]:
+) -> Generator[list[Service], Any, Any]:
+    num_resources = getattr(request, "param", {}).get("num", 1)
     if pytestconfig.option.post_upgrade:
-        mr_db_service = Service(name=DB_RESOURCES_NAME, namespace=model_registry_namespace, ensure_exists=True)
-        yield mr_db_service
+        mr_db_service = Service(name=DB_RESOURCE_NAME, namespace=model_registry_namespace, ensure_exists=True)
+        yield [mr_db_service]
         mr_db_service.delete(wait=True)
     else:
-        with Service(
+        services = get_mr_service_objects(
```

Comment:
why not use [NamespacedResourceList](https://github.com/RedHatQE/openshift-python-wrapper/blob/main/ocp_resources/resource.py#L1970)?
should make everything more simple